lib/inv_cg_quda.cpp

```diff
@@ -244,23 +252,29 @@ namespace quda {
 
     if (!param.is_preconditioner) getProfile().TPSTART(QUDA_PROFILE_INIT);
 
-    double b2 = blas::norm2(b);
+    vector<double> b2 = blas::norm2(b);
 
     // Check to see that we're not trying to invert on a zero-field source
-    if (b2 == 0 && param.compute_null_vector == QUDA_COMPUTE_NULL_VECTOR_NO) {
-      if (!param.is_preconditioner) getProfile().TPSTOP(QUDA_PROFILE_INIT);
-      printfQuda("Warning: inverting on zero-field source\n");
-      x = b;
-      param.true_res = 0.0;
-      param.true_res_hq = 0.0;
-      return;
+    if (param.compute_null_vector == QUDA_COMPUTE_NULL_VECTOR_NO) {
+      bool zero_src = true;
+      for (auto i = 0u; i < b.size(); i++) {
+        if (b2[i] == 0) {
+          warningQuda("inverting on zero-field source");
```

Comment:
I'd add the index of the zero-field source, so more like `warningQuda("inverting on zero-field source %d", i);` or something of that ilk.

Reply:
f0d9f3c8dc79bdb4ed751b34193262c517a6b287